src/binary-reader.cc

```diff
@@ -944,8 +956,22 @@ Result BinaryReader::ReadFunctionBody(Offset end_offset) {
         CHECK_RESULT(ReadAlignment(&alignment_log2, "load alignment"));
         Address offset;
         CHECK_RESULT(ReadAddress(&offset, 0, "load offset"));
-        CALLBACK(OnLoadExpr, opcode, alignment_log2, offset);
-        CALLBACK(OnOpcodeUint32Uint32, alignment_log2, offset);
+        Index memidx;
+        if (alignment_log2 >> 6) {
+          ERROR_IF(!options_.features.multi_memory_enabled(),
+              "multi_memory not allowed");
+          CHECK_RESULT(ReadMemidx(&memidx, "store memidx"));
+          alignment_log2 = alignment_log2 & ((1 << 6) - 1);
+        } else {
+          memidx = 0;
+        }
+        
+        CALLBACK(OnLoadExpr, opcode, alignment_log2, offset, memidx);
+        if (memidx == 0) {
+          CALLBACK(OnOpcodeUint32Uint32, alignment_log2, offset);
+        } else {
+          CALLBACK(OnOpcodeUint32Uint32Uint32, alignment_log2, offset, memidx);
```

Comment:
Oh I see.. since objdump doesn't know it can elide the memidx when dumping the instruction.     I guess this is OK for now and we can look into making objdump smart as a potential followup.

Reply:
Here is an example of the test output when we only keep OnOpcodeUint32Uint32Uint32.
>   test/opcodecnt/basic.txt                                                       
>   STDOUT MISMATCH:
>   --- expected
>   +++ actual
>   @@ -12,7 +12,7 @@
>    end: 2
>    br 0: 2
>    local.get 1: 2
>   -i32.load 2, 0: 1
>   +i32.load 2, 0, 0: 1
>    i32.const 0 (0x0): 1
>    i32.const 1 (0x1): 1
>    i32.const 2 (0x2): 1